### DISCUSSION TITLE
read the files from assets or public folders

### DISCUSSION THREAD

**Discussion by ostapoff** [nuxt/nuxt#16438] [Questions]
Hi guys,

I need to generate a PDF file in the API call handler located in the /server/api. The PDF should contain some images then from the /assets or /public folder. Is there any legal way to resolve a path to the images to read them by the server process?

**Comment by zernonia** [comment]
Following this too. I have some text file needed to pull from `/assets` or `/public` into server route as well. Not sure how to resolve it properly

**Comment by mishushakov** [comment]
Hey, I had a similar issue, here's what worked for me:

1. Put your files into `/public` folder
2. In your server:

  ```js
  import path from 'path'
  ```
  
  Locate the file using `path` module
  
  ```js
  path.resolve('public/file.txt')
  ```

**Comment by KabooZ1987** [comment]
I'm having the same issue and no probable solution I tried worked...  

resolving the path sadly also didn't work  
the resulting path is always `.../.output/server/public/...`

**Reply by KabooZ1987** [reply]
though I could make it work with Environment variables  

```js
process.env.NODE_ENV == "production" ? '../public/data' : 'public/data/'
```
